testnet/tools/nns-tools/list-new-commits.sh

```diff
@@ -19,39 +19,98 @@ Usage: $0 (<COMMIT_ID>)
     exit 1
 }
 
-if [ $# -gt 1 ]; then
-    help
+RELEASE_CANDIDATE_COMMIT_ID="${1:-GARBAGE}"
+
+AUTO_COMMIT_ID=false
+if [[ "${RELEASE_CANDIDATE_COMMIT_ID}" == 'GARBAGE' ]]; then
+    RELEASE_CANDIDATE_COMMIT_ID=$(latest_commit_with_prebuilt_artifacts 2>/dev/null)
+    AUTO_COMMIT_ID=true
+
+    COMMIT_DESCRIPTION=$(git show -s --format="%h (%cd)" --date=relative "${RELEASE_CANDIDATE_COMMIT_ID}")
+    echo
+    print_yellow "⚠️  Using the latest commit with prebuilt artifacts: ${COMMIT_DESCRIPTION}"
 fi
 
-LATEST_ARTIFACTS_COMMIT=$(latest_commit_with_prebuilt_artifacts 2>/dev/null)
+NETWORK=ic
+
+list_new_canister_commits() {
+    CANISTER_NAME="${1}"
+    # Notice plural. Multiple values are separated by space. Because bash.
+    CODE_DIRECTORIES="${2}"
+    LATEST_RELEASED_COMMIT_ID="${3}"
+
+    RANGE="${LATEST_RELEASED_COMMIT_ID}..${RELEASE_CANDIDATE_COMMIT_ID}"
+    NEW_COMMITS=$(
+        git \
+            --no-pager \
+            log \
+            --format="%C(auto) %h %s" \
+            "${RANGE}" \
+            -- \
+            ${CODE_DIRECTORIES} # No quote here is necessary, because plural.
+    )
 
-RELEASE_CANDIDATE_COMMIT_ID=${1:-$LATEST_ARTIFACTS_COMMIT}
-echo "Listing commits from:" "$RELEASE_CANDIDATE_COMMIT_ID"
+    INTERESTING_COMMITS=$(
+        grep -v -E ' .{10} (chore|refactor|test)\b' <<<"$NEW_COMMITS" \
```

Comment:
I'm a little wary, in this script, of filtering these out.  It assumes we're all correctly labelling every PR every time, and sometimes there's some interpretation about what conventional commit prefix to use.   I'm wondering if maybe highlighting interesting commits with a different color would be better, so that they pop out?

Reply:
Yes, highlighting good. I thought of that, but I could not come up with a simple way to do it. Do you have any recommendations?

If a commit has wrong prefix, I don't think showing it is really going to compensate, because the way I look at these commit lists is to (manually) scan the prefix. This just automates what I already do manually. Unless others are actually reading the entire title (including the ones with boring prefixes), I think it is better to just not show those, because we want to maintain a high signal-to-noise ratio.

Reply:
For now, I'm going to go with what I have, but please, feel free to propose more PRs.